code/modules/reagents/reagent_containers/borghydro.dm

```diff
@@ -48,82 +70,57 @@
 	if(charge_tick < recharge_time)
 		return FALSE
 	charge_tick = 0
-
-	if(isrobot(loc))
-		var/mob/living/silicon/robot/R = loc
-		if(R && R.cell)
-			var/datum/reagents/RG = reagent_list[mode]
-			if(!refill_borghypo(RG, reagent_ids[mode], R)) 	//If the storage is not full recharge reagents and drain power.
-				for(var/i in 1 to reagent_list.len)     	//if active mode is full loop through the list and fill the first one that is not full
-					RG = reagent_list[i]
-					if(refill_borghypo(RG, reagent_ids[i], R))
-						break
-	//update_icon()
+	refill_borghypo()
 	return TRUE
 
 // Use this to add more chemicals for the borghypo to produce.
-/obj/item/reagent_containers/borghypo/proc/add_reagent(reagent)
-	reagent_ids |= reagent
-	var/datum/reagents/RG = new(30)
-	RG.my_atom = src
-	reagent_list += RG
-
-	var/datum/reagents/R = reagent_list[reagent_list.len]
-	R.add_reagent(reagent, 30)
-
-/obj/item/reagent_containers/borghypo/proc/refill_borghypo(datum/reagents/RG, reagent_id, mob/living/silicon/robot/R)
-	if(RG.total_volume < RG.maximum_volume)
-		RG.add_reagent(reagent_id, BORGHYPO_REFILL_VALUE)
-		R.cell.use(charge_cost)
-		return TRUE
-	return FALSE
+/obj/item/reagent_containers/borghypo/proc/refill_borghypo(reagent_id, mob/living/silicon/robot/robot)
+	if(istype(robot))
+		robot.cell.use(charge_cost)
```

Comment:
I'll give it a once over when I get home, eta 8 hours. 
It shouldn't runtime because your modules get yeeted if you got no cell, but can't ever be too sure

Reply:
Fine in testing